feature_layers/toggle-between-feature-request-modes/src/main/java/com/esri/samples/toggle_between_feature_request_modes/ToggleBetweenFeatureRequestModesSample.java

```diff
@@ -0,0 +1,242 @@
+      mapView.setMap(map);
+      // set the starting viewpoint for the map view
+      mapView.setViewpoint(new Viewpoint(45.5266, -122.6219, 6000));
+      // show a progress indicator when the map view is drawing (e.g. when fetching caches)
+      mapView.addDrawStatusChangedListener(e -> progressIndicator.setVisible(e.getDrawStatus() == DrawStatus.IN_PROGRESS));
+
+      // add label and button to the control panel
+      controlsVBox.getChildren().addAll(cacheButton, noCacheButton, manualCacheButton, label, populateButton);
+
+      // add the map view and control panel to stack pane
+      stackPane.getChildren().addAll(mapView, controlsVBox, progressIndicator);
+      StackPane.setAlignment(controlsVBox, Pos.TOP_LEFT);
+      StackPane.setMargin(controlsVBox, new Insets(10, 0, 0, 10));
+    } catch (Exception e) {
+      // on any error, display the stack trace
+      e.printStackTrace();
+    }
+  }
+
+  /**
+   * Fetches the cache from a Service Feature Table manually.
+   */
+  private void fetchCacheManually() {
+
+    // create query to select all tree features
+    QueryParameters queryParams = new QueryParameters();
+    // query for all tree conditions except "dead" with coded value '4' within the visible extent
+    queryParams.setWhereClause("Condition < '4'");
+    queryParams.setGeometry(mapView.getVisibleArea().getExtent());
+
+    List<String> outfields = Collections.singletonList("*");     // * means all features
+    // get queried features from service feature table and clear previous cache
+    ListenableFuture<FeatureQueryResult> tableResult = featureTable.populateFromServiceAsync(queryParams, true, outfields);
+
+    tableResult.addDoneListener(() -> {
+      try {
+        // find the number of features returned from query
+        AtomicInteger featuresReturned = new AtomicInteger();
+        tableResult.get().forEach(feature -> featuresReturned.getAndIncrement());
+
+        // display to user how many features where returned
+        // note the service has a maximum record count of 2000
+        label.setText("Populated " + featuresReturned + " features.");
+      } catch (Exception e) {
+        // on any error, display the stack trace
+        e.printStackTrace();
+      }
+    });
+  }
+
+  /**
+   * Sets up control panel with radio buttons to toggle between feature request modes, a label displaying the features
+   * returned from a manual cache result, and a button to manually request cache if manual cache feature request mode
+   * is selected.
+   */
+  private void setUpUi() {
+
+    // create a control panel
+    controlsVBox = new VBox(6);
+    controlsVBox.setBackground(new Background(new BackgroundFill(Paint.valueOf("rgba(0,0,0,0.5)"), CornerRadii.EMPTY,
+      Insets.EMPTY)));
+    controlsVBox.setPadding(new Insets(10.0));
+    controlsVBox.setMaxSize(200, 80);
+    controlsVBox.getStyleClass().add("panel-region");
+
+    // create a label to display number of features returned
```

Comment:
```suggestion
    // create a label to display information on the UI
```
It has multiple purposes